src/components/Capture/withCrossDeviceWhenNoCamera.js

```diff
@@ -8,20 +8,22 @@ export default WrappedComponent =>
     }
 
     componentDidUpdate(prevProps) {
-      const propsWeCareAbout = ["useWebcam", "hasCamera", "allowCrossDeviceFlow", "forceCrossDevice"]
-      const propsHaveChanged = propsWeCareAbout
-        .some(propKey => prevProps[propKey] !== this.props[propKey])
+      const propsWeCareAbout = ["currentStep", "hasCamera", "allowCrossDeviceFlow", "forceCrossDevice"]
+      const propsHaveChanged = propsWeCareAbout.some(propKey => prevProps[propKey] !== this.props[propKey])
 
-      if (propsHaveChanged) {
-        this.attemptForwardToCrossDevice(this.props)
+      if (propsHaveChanged && this.props.allowCrossDeviceFlow) {
+        this.attemptForwardToCrossDevice()
       }
     }
 
-    attemptForwardToCrossDevice = (props = this.props) => {
-      const { useWebcam, hasCamera, forceCrossDevice, allowCrossDeviceFlow, changeFlowTo } = props
-      const shouldUseCamera = useWebcam && hasCamera === false
-
-      if (allowCrossDeviceFlow && (shouldUseCamera || forceCrossDevice)) {
+    attemptForwardToCrossDevice = () => {
+      const { hasCamera, forceCrossDevice, changeFlowTo, componentsList, step } = this.props
+      const currentStep = componentsList[step]
+      const cameraRequiredButNoneDetected = currentStep.step.type === 'face' && !hasCamera
```

Comment:
Isn't there access to `method` in this component?

Reply:
As discussed, did not see any prop named `method` in the component and this was the only way to get the `currentStep` that SDK is on.